src/Symfony/Component/Config/Definition/Builder/ExprBuilder.php

```diff
@@ -61,7 +61,7 @@ public function always(?\Closure $then = null): static
      */
     public function ifTrue(?\Closure $closure = null): static
     {
-        $this->ifPart = $closure ?? static fn ($v) => true === $v;
+        $this->ifPart = $closure ?? static fn ($v) => (bool) $v;
```

Comment:
here, we don't strictly compare the result of the closure when one is passed, so let's keep the default part and be consistent below:
```suggestion
        $this->ifPart = $closure ?? static fn ($v) => true === $v;
```